og.module

```diff
@@ -154,6 +155,21 @@ function og_entity_access(EntityInterface $entity, $operation, AccountInterface
   /** @var \Drupal\Core\Access\AccessResult $access */
   $access = \Drupal::service('og.access')->userAccessEntity($operation, $entity, $account);
 
+  $audience_fields = \Drupal::service('og.group_audience_helper')->getAllGroupAudienceFields($entity_type_id, $bundle_id);
+  if (count($audience_fields) === 1) {
+    $field_definitions = \Drupal::service('entity_field.manager')->getFieldDefinitions($entity_type_id, $bundle_id);
+    foreach ($field_definitions as $field_definition) {
+
+      /** @var \Drupal\Core\Field\FieldDefinitionInterface $field_definition */
+      if (!in_array($field_definition->getName(), array_keys($audience_fields))) {
+        continue;
+      }
+
+      // Check if the field needs to be required.
+      $field_definition->setRequired($access);
```

Comment:
@MPParsley this one is related to https://github.com/Gizra/og/pull/570/files#r377001979

We try to make sure that if a node must be created inside an OG context, it cannot be created without choosing a group. Again not sure about the correctness of code - but rather on the reason.

Reply:
I think this code is not working. This is calling `FieldConfigInterface::setRequired()` but never calling `FieldConfigInterface::save()` after making the change so this code is basically doing nothing.

We should not alter field definitions in `hook_entity_access()` in any case.